### ISSUE TITLE
Highlight JSX in plain old js2-mode

### ISSUE THREAD

**Issue by jacksonrayhamilton** [mooz/js2-mode#528]
JSX highlighting is currently available only if

- the user installs Emacs 27 and uses js-mode with js2-minor-mode enabled, or
- the user installs [rjsx-mode](https://github.com/felipeochoa/rjsx-mode).

It would be nice if JSX was also highlighted when plain ol’ js2-mode is enabled.

As we decided in https://github.com/mooz/js2-mode/pull/523, the highlighting that js-mode does won’t be inherited by js2-mode, because js2-mode highlights code based on its AST, not with font-lock-keywords.

Therefore, should we ever parse JSX nodes and add them to the AST (as proposed in https://github.com/mooz/js2-mode/issues/527), then during that parsing, we should also record the locations where faces should be applied to JSX so that it’s highlighted as such.  It should be highlighted like SGML, just like js-mode does it in Emacs 27.  This issue proposes such an improvement to js2-mode.

**Comment by dgutov** [comment]
Sure. Except the current `js2-mode` parser doesn't read the nodes in good enough detail to do that. `rjsx-mode` already does that, I think simply during parsing.

So I think this issue is a subset of 527.

**Comment by taigacute** [comment]
It does seem to have good support in emacs28, but the highlighting seems to be missing.
![image](https://user-images.githubusercontent.com/41671631/76165466-3de8ae00-6192-11ea-80e1-342600b53db9.png)

**Comment by dgutov** [comment]
Highlighting of..?

**Comment by taigacute** [comment]
syntax highlight .open js2-mode it will like this.
![image](https://user-images.githubusercontent.com/41671631/76180888-44127500-61fa-11ea-8368-cd4803c836c0.png)

**Comment by dgutov** [comment]
Hmm, yes. That's correct. `js-mode` is a lot more subtle in syntax highlighting.

But that's a historical thing. You can file a feature request for it to highlight more things.

**Comment by taigacute** [comment]
I use the [color-indentifilers-mode](https://github.com/ankurdave/color-identifiers-mode`) with js-mode  on emacs28 got a good syntax highlight
![image](https://user-images.githubusercontent.com/41671631/76515979-86a3ae00-6495-11ea-8dbb-69a39af2b4a6.png)

**Comment by jacksonrayhamilton** [comment]
@taigacute, this issue is about highlighting JSX syntax (and only that syntax) in js2-mode, _in the absence of other software_.  It’s not seeking the shortest path to a personal solution, or any alternatives or workarounds for any other scenarios.

We’re glad you found something that works for you, but this is not the place to the post it, unless it is relevant to the implementation proposed in the issue description.

**Comment by dgutov** [comment]
Indeed. Sorry, I also played a part in side-tracking this discussion.
